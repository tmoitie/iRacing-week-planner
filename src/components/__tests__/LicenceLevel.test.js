import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { describe, expect, test } from '@jest/globals';

import LicenceLevel from '../LicenceLevel';

describe('components/LicenceLevel', () => {
  test('renders for high D Class', () => {
    const { container } = render(<LicenceLevel licence={8} />);
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveTextContent('D');
    expect(container.firstChild).toHaveTextContent('4.00');
    expect(container.firstChild).toHaveClass('licenceD');
  });

  test('renders for Pro', () => {
    const { container } = render(<LicenceLevel licence={21} />);
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveTextContent('1.00');
    expect(container.firstChild).toHaveClass('licenceP');
  });

  test('renders for float levels', () => {
    const { container } = render(<LicenceLevel licence={7.5} />);
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveTextContent('D');
    expect(container.firstChild).toHaveTextContent('3.50');
    expect(container.firstChild).toHaveClass('licenceD');
  });

  test('renders for other float levels', () => {
    const { container } = render(<LicenceLevel licence={8.5} />);
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveTextContent('D');
    expect(container.firstChild).toHaveTextContent('4.50');
    expect(container.firstChild).toHaveClass('licenceD');
  });

  test('renders effective licence', () => {
    const { container } = render(<LicenceLevel licence={8} effective />);
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveTextContent('C');
    expect(container.firstChild).not.toHaveTextContent('3.00');
    expect(container.firstChild).toHaveClass('licenceC');
  });
});
