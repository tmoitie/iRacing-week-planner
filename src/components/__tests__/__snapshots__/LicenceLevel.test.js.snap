// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LicenceLevel renders effective licence 1`] = `
<div>
  <div
    class="licenceLevelComponent licenceC"
  >
    <span
      class="licenceLetter"
    >
      C
    </span>
    <span
      class="licenceText"
    />
  </div>
</div>
`;

exports[`components/LicenceLevel renders for Pro 1`] = `
<div>
  <div
    class="licenceLevelComponent licenceP"
  >
    <span
      class="licenceLetter"
    >
      P
    </span>
    <span
      class="licenceText"
    >
      1.00
    </span>
  </div>
</div>
`;

exports[`components/LicenceLevel renders for float levels 1`] = `
<div>
  <div
    class="licenceLevelComponent licenceD"
  >
    <span
      class="licenceLetter"
    >
      D
    </span>
    <span
      class="licenceText"
    >
      3.50
    </span>
  </div>
</div>
`;

exports[`components/LicenceLevel renders for high D Class 1`] = `
<div>
  <div
    class="licenceLevelComponent licenceD"
  >
    <span
      class="licenceLetter"
    >
      D
    </span>
    <span
      class="licenceText"
    >
      4.00
    </span>
  </div>
</div>
`;

exports[`components/LicenceLevel renders for other float levels 1`] = `
<div>
  <div
    class="licenceLevelComponent licenceD"
  >
    <span
      class="licenceLetter"
    >
      D
    </span>
    <span
      class="licenceText"
    >
      4.50
    </span>
  </div>
</div>
`;
